Frost mage is ready to go

syrifgit opened the pull request:
Beating dummy sims with both hero trees

syrifgit committed:
Frost mage is ready to go

Beating dummy sims with both hero trees

syrifgit commented:
Amendment is coming actually, no push.

Hekili commented on TheWarWithin/Priorities/MageFrost.simc:
```diff
@@ -1,24 +1,27 @@
-actions+=/run_action_list,name=cleave,strict=1,if=active_enemies>=2&active_enemies<=3
-actions+=/run_action_list,name=ss_st,strict=1,if=talent.splinterstorm
-actions+=/run_action_list,name=st
+actions+=/call_action_list,name=aoe,strict=1,if=active_enemies>=7|active_enemies>=3&!talent.frostfire_bolt|active_enemies>=4
```
I'm confused; the first `active_enemies>=7|` is redundant because `|active_enemies>=4` will also be true any time you have 7+ targets.

Is there some grouping to these clauses that is intended?  If the intent is just that you AOE with 3+ targets without Frostfire Bolt talented or 4+ targets with Frostfire Bolt, this simplifies to:

`active_enemies>2&!talent.frostfire_bolt|active_enemies>3`

Note that I changed `>=3` to `>2` and `>=4` to `>3`.  The difference might save a whole nanosecond per year.

syrifgit replied:
I think this condition might be changing anyways with the `actions.aoe` change I am submitting

syrifgit replied:
that or just a silly entry by the APL writer, will simplify.

Hekili commented on TheWarWithin/Priorities/MageFrost.simc:
```diff
@@ -1,24 +1,27 @@
+actions+=/call_action_list,name=cleave_ss,strict=1,if=active_enemies>=2
+actions+=/call_action_list,name=st_aoebuild,strict=1,if=talent.splinterstorm&!(talent.cold_front&talent.slick_ice&talent.deaths_chill&talent.frozen_touch)|talent.frostfire_bolt&!talent.deep_shatter
+actions+=/call_action_list,name=st_ff,strict=1,if=talent.frostfire_bolt
+actions+=/call_action_list,name=st_ss
```
We could save some CPU cycles by making this one `strict=1,if=!talent.frostfire_bolt` unless you want `st_ss` list items tested after `st_ff`.

syrifgit replied:
st_ff probably shouldn't ever completely fail, but will add for consistency.

Hekili commented on TheWarWithin/Priorities/MageFrost.simc:
```diff
@@ -29,12 +32,28 @@ actions.aoe+=/cone_of_cold,if=target.within10&!talent.coldest_snap&active_enemie
 
+actions.aoe+=/cone_of_cold,if=talent.coldest_snap&cooldown.frozen_orb.remains>4&(prev_gcd.1.comet_storm|prev_gcd.1.frozen_orb&!talent.comet_storm|cooldown.comet_storm.remains>15&!talent.frostfire_bolt)
+actions.aoe+=/frozen_orb,if=cooldown_react&((!prev_gcd.1.cone_of_cold|!talent.isothermic_core)&(!prev_gcd.1.glacial_spike|!freezable))
+actions.aoe+=/blizzard,if=!prev_gcd.1.glacial_spike|!freezable
```
We should think about the different prev/prev_gcd entries some more, though this doesn't keep us from moving forward with this PR.  There can be a difference between "the last GCD spell was X" vs. "X was used <= 1 GCD ago" (or if you want to relax the requirement in case somebody hits another GCD spell.

syrifgit replied:
I can definitely revist on the next pass. Sometimes it's because you need to spellqueue (anytime flurry is involved) which would be fine for time since, sometimes it's just "this spell must be after that spell".

Hekili commented on TheWarWithin/Priorities/MageFrost.simc:
```diff
@@ -46,64 +65,72 @@ actions.cds+=/lights_judgment
-actions.ss_st+=/call_action_list,name=movement
+actions.st_aoebuild+=/comet_storm,if=prev_gcd.1.flurry&(buff.icy_veins.down|talent.frostfire_bolt)
+actions.st_aoebuild+=/flurry,if=cooldown_react&(buff.icicles.stacks<5|talent.splinterstorm)&(remaining_winters_chill=0&debuff.winters_chill.down&(prev_gcd.1.frostbolt|prev_gcd.1.frostfire_bolt|prev_gcd.1.glacial_spike)|buff.excess_frost.react)
+actions.st_aoebuild+=/frozen_orb,if=cooldown_react&(talent.splinterstorm|(!talent.ray_of_frost|buff.fingers_of_frost.down&cooldown.ray_of_frost.remains&buff.icicles.stacks<5))
```
Not sure if both `stacks` and `stack` work, but I think they do.

Hekili commented on TheWarWithin/Priorities/MageFrost.simc:
```diff
@@ -46,64 +65,72 @@ actions.cds+=/lights_judgment
-actions.ss_st+=/frostbolt
-actions.ss_st+=/call_action_list,name=movement
+actions.st_aoebuild+=/comet_storm,if=prev_gcd.1.flurry&(buff.icy_veins.down|talent.frostfire_bolt)
+actions.st_aoebuild+=/flurry,if=cooldown_react&(buff.icicles.stacks<5|talent.splinterstorm)&(remaining_winters_chill=0&debuff.winters_chill.down&(prev_gcd.1.frostbolt|prev_gcd.1.frostfire_bolt|prev_gcd.1.glacial_spike)|buff.excess_frost.react)
```
See previous discussion of `.react`.

syrifgit replied:
![image](https://github.com/user-attachments/assets/4986a8b4-80a2-4f7d-84ac-2915df6cd9ad)

In my forthcoming revision

syrifgit committed:
Frost Mage

Review notes implemented, `actions.aoe` reworked as per discussion with Frost Mage Theorycrafter. Added comment for the flurry time<0.2 so it doesn't get left behind again. All reacts are replaced with the appropriate buff or cd checks.

syrifgit committed:
Frost mag

Accompanying pack string, remove unneeded variable, allow pet freeze to be cast while casting

syrifgit commented:
@Hekili ready for realsies
